src/librbd/io/ImageDispatcher.cc

```diff
@@ -26,6 +24,67 @@
 namespace librbd {
 namespace io {
 
+template <typename I>
+struct ImageDispatcher<I>::C_LayerIterator : public Context {
```

Comment:
Nit: can this be pulled into a common helper in `librbd/io/Utils.h` and be re-used between the object- and image-dispatchers?

Reply:
I moved it to librbd/io/Dispather.*.